include/deal.II/base/data_out_base.h

```diff
@@ -1383,9 +1383,17 @@ namespace DataOutBase
       bool operator() (const Point<3> &point_one,
                        const Point<3> &point_two) const
       {
-        return (point_one(0) < point_two(0)
-                || (!(point_two(0) < point_one(0)) && (point_one(1) < point_two(1)
-                                                       || (!(point_two(1) < point_one(1)) && point_one(2) < point_two(2)))));
+        const double eps = 100.0 * std::numeric_limits<double>::epsilon();
+
+        for (unsigned int d=0; d<3; ++d)
+          {
+            if (point_one(d) < point_two(d) - eps * std::abs(point_two(d)))
+              return true;
+            else if ( d==2 || point_one(d) > point_two(d) + eps * std::abs(point_two(d)))
```

Comment:
Why do you have the extra check for `d==2`? That case should return the `false` at the end of the function in line 1396, right? To avoid confusion, I would omit this check - I was looking for an invalid access somewhere... ;-)

Reply:
Yes that is not longer necessary. I copied it from @davydden's suggestion (where it made sense), but had to add the `return false` at the end to avoid a warning, so no need for it anymore. Fixed.